source/module_esolver/esolver_ks_lcao_elec.cpp

```diff
@@ -321,6 +323,41 @@ void ESolver_KS_LCAO<TK, TR>::before_scf(int istep)
 #endif // __EXX
 
     this->pelec->init_scf(istep, this->sf.strucFac);
+    if(GlobalV::out_chg == 2)
```

Comment:
Move the details of output charge density to module_io, you can add a new class / source file to store these detailed options, otherwise the Esolver source file will be super large. The esolver is not a place that stores all of the output details.